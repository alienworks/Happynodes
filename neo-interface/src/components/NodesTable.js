import React, { Component } from 'react'
import { connect } from 'react-refetch'
import config from './config'

class NodesTable extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { nodes } = this.props

        if (nodes.pending) {
            return (
                <div class="well"> Loading </div>
            );
        }
        return (
            <div class="container">
                <table class="table">
                    <thead>
                        <tr>
                            <th>Endpoint</th>
                            <th>Type</th>
                            <th>Blockheight</th>
                            <th>Mempool Size</th>
                            <th>Latency (ms)</th>
                            <th>Block Height</th>
                            <th>Peers</th>
                            <th>Stability</th>
                            <th>Version</th>
                        </tr>
                    </thead>
                    <tbody>
                        {nodes.value.map((node, i) =>
                            <tr key={i}>
                                <td>{node.address}</td>
                                <td>RPC</td>
                                <td>{node.blockheight}</td>
                                <td>{node.mempool_size}</td>
                                <td>{node.latency}</td>
                                {node.rpc_http_status === true &&
                                    <td>Okay</td>
                                }
                                {node.rpc_http_status === false &&
                                    <td>N/A</td>
                                }
                                {node.rpc_http_status === true &&
                                    <td>Okay</td>
                                }
                                {node.rpc_http_status === false &&
                                    <td>N/A</td>
                                }
                                {node.online === true &&
                                    <td>Online</td>
                                }
                                {node.online === false &&
                                    <td>Offline</td>
                                }
                                <td>{node.rpc_http_status}</td>
                                <td>{node.rpc_https_status}</td>
                                <td>{node.connection_counts}</td>
                                <td>{node.count} pct</td>
                                <td>{node.version}</td>
                            </tr>
                        )}
                    </tbody>
                </table>

            </div>
        );
    }
}

export default connect((props) => ({
    nodes: {
        url: config.api_url.concat(`/nodes`),
        refreshInterval: 5000
    }
}))(NodesTable)