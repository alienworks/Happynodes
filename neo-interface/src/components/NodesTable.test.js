import React from 'react';
import ReactDOM from 'react-dom';
import NodesTable from './NodesTable';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NodesTable />, div);
    ReactDOM.unmountComponentAtNode(div);
});

var data = [  
       {  
          "id":37,
          "hostname":"pyrpc2.narrative.org",
          "protocol":"https",
          "port":443,
          "p2p_tcp_status":false,
          "p2p_ws_status":false,
          "address":"https://pyrpc2.narrative.org:443",
          "validated_peers_counts":29,
          "stability":100,
          "blockheight_score":96,
          "normalised_latency_score":87.1326684951782,
          "validated_peers_counts_score":59.183673469387756,
          "health_score":85.5790854911415,
          "latency":257.346630096436,
          "rcp_https_status":true,
          "rcp_http_status":true,
          "mempool_size":15,
          "connection_counts":116,
          "online":true,
          "blockheight":2624393,
          "lat":40.7904,
          "long":-74.0247,
          "locale":"us",
          "version":"/NEO-PYTHON:0.7.6/",
          "max_blockheight":2624395
       }
    ]

describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<NodesTable nodes={{value:data}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});