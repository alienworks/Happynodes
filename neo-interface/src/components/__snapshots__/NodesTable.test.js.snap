// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  class="container"
>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Endpoint
        </th>
        <th>
          Type
        </th>
        <th>
          Blockheight
        </th>
        <th>
          Mempool Size
        </th>
        <th>
          Latency (ms)
        </th>
        <th>
          Block Height
        </th>
        <th>
          Peers
        </th>
        <th>
          Stability
        </th>
        <th>
          Version
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          https://pyrpc2.narrative.org:443
        </td>
        <td>
          RPC
        </td>
        <td>
          2624393
        </td>
        <td>
          15
        </td>
        <td>
          257.346630096436
        </td>
        <td>
          Online
        </td>
        <td />
        <td />
        <td>
          116
        </td>
        <td>
           pct
        </td>
        <td>
          /NEO-PYTHON:0.7.6/
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
